tests/test_h5store.py

```diff
@@ -103,18 +103,22 @@ def test_open_read_only(self):
             h5s["foo"] = "bar"
 
         with self.open_h5store(mode="r") as h5s:
+            assert h5s.mode == "r"
             assert "foo" in h5s
             self.assertEqual(h5s["foo"], "bar")
 
     def test_open_write_only(self):
         with self.open_h5store(mode="w") as h5s:
+            assert h5s.mode == "r+"
             h5s["foo"] = "bar"
             assert "foo" in h5s
             self.assertEqual(h5s["foo"], "bar")
 
     def test_open_write_and_read_only(self):
         with self.open_h5store(mode="w") as h5s_w:
             with self.open_h5store(mode="r") as h5s_r:
+                assert h5s_w.mode == "r+"
+                assert h5s_r.mode == "r+"
```

Comment:
I expected this mode to be `r` instead of `r+`. I did an experiment with swapping the order of these lines. I found it's not possible (on Linux, at least) to open a writeable handle to the file if the read-only handle is created first. To me, this calls into question whether this test is actually useful: the file handle that we're claiming is read-only is NOT read-only -- it is actually using the same file handle as the writeable store. Maybe this doesn't matter. I don't know. 🤷‍♂️

Reply:
We could potential raise a warning in this case if we know that the store is open currently in a writable mode. However, that doesn't need to be in this PR. I would personally remove the test, as it doesn't test anything (well the behavior it is meant to test doesn't work).

Reply:
I see what you mean. Let's leave the test for now, because if this behavior of the upstream library ever changed or varied by platform, we would like to know.